scalanet/ut/src/io/iohk/scalanet/peergroup/DynamicTLSPeerGroupSpec.scala

```diff
@@ -60,6 +65,33 @@ class DynamicTLSPeerGroupSpec extends AsyncFlatSpec with BeforeAndAfterAll {
     }
   }
 
+  it should "handshake successfully via proxy" in taskTestCase {
+    (for {
+      client <- DynamicTLSPeerGroup[String](getCorrectConfig())
+      proxy <- buildProxyServer(1080)
+      server <- DynamicTLSPeerGroup[String](getCorrectConfig())
+      ch1 <- client.client(server.processAddress, Socks5Config(new InetSocketAddress("localhost", 1080), None))
+    } yield (server, ch1, proxy)).use {
+      case (server, ch1, proxy) =>
+        for {
+          _ <- ch1.sendMessage("Hello enc server")
+          wo <- server.serverEventObservable.collectChannelCreated.mergeMap {
+            case (ch, release) =>
+              Observable(ch).zip(ch.channelEventObservable.guarantee(release))
+
+          }.headL
+          sessions <- Task(proxy.getSessionManager.getManagedSessions)
+          (chan, rec) = wo
+        } yield {
+          assert(sessions.size() == 1)
+          val session = sessions.get(1.toLong)
+          assert(session.getNetworkMonitor.getReceiveTCP > 0)
+          assert(session.getNetworkMonitor.getSendTCP > 0)
```

Comment:
Is `getReceiveTCP` tracking what `ch1` sent to the proxy, and `getSendTCP` what the proxy sent  to the `server`? Is there a separate count for responses coming from the `server`?

Reply:
thats my guess, but the library does not have great docs so it is as good as yours. From what i known that all are stats for tcp we can get. There is also the same pair of stats for udp.